base_layer/core/src/base_node/tari_pulse_service/mod.rs

```diff
@@ -0,0 +1,216 @@
+        }
+    }
+
+    async fn get_node_block(
+        &mut self,
+        base_node_service: &mut LocalNodeCommsInterface,
+        block_height: u64,
+    ) -> Result<(u64, String), anyhow::Error> {
+        let historical_block = base_node_service
+            .get_header(block_height)
+            .await
+            .and_then(|header| match header {
+                Some(header) => Ok((header.height(), header.hash().to_hex())),
+                None => {
+                    error!(target: LOG_TARGET, "Header not found for height: {}", block_height);
+                    Err(CommsInterfaceError::InternalError("Header not found".to_string()).into())
+                },
+            })?;
+
+        Ok(historical_block)
+    }
+
+    pub async fn fetch_checkpoints(&mut self) -> Result<Vec<(u64, String)>, anyhow::Error> {
+        let mut client = self.get_dns_client().await?;
+        let query = client.query(self.dns_name.clone(), DNSClass::IN, RecordType::TXT);
+        let response = query.await?;
+        let answers: &[Record] = response.answers();
+        let checkpoints: Vec<(u64, String)> = answers
+            .iter()
+            .filter_map(|record| {
+                if let RData::TXT(txt) = record.data() {
+                    let ascii_txt = txt.txt_data().iter().fold(String::new(), |mut acc, bytes| {
+                        acc.push_str(&String::from_utf8_lossy(bytes));
+                        acc
+                    });
+                    let (height, hash) = ascii_txt.split_once(':')?;
+                    return Some((height.parse().ok()?, hash.to_string()));
+                }
+                None
+            })
+            .collect();
+
+        Ok(checkpoints)
+    }
+}
+
+#[derive(Clone)]
+pub struct TariPulseHandle {
+    pub shutdown_signal: ShutdownSignal,
+    pub failed_checkpoints_notifier: watch::Receiver<bool>,
+}
+
+impl TariPulseHandle {
+    pub fn get_failed_checkpoints_notifier(&self) -> watch::Ref<'_, bool> {
+        self.failed_checkpoints_notifier.borrow()
+    }
+}
+
+pub struct TariPulseServiceInitializer {
+    interval: Option<Duration>,
+    network: Network,
+}
+
+impl TariPulseServiceInitializer {
+    pub fn new(interval: Option<Duration>, network: Network) -> Self {
+        Self { interval, network }
+    }
+}
+
+#[async_trait]
+impl ServiceInitializer for TariPulseServiceInitializer {
+    async fn initialize(&mut self, context: ServiceInitializerContext) -> Result<(), ServiceInitializationError> {
+        info!(target: LOG_TARGET, "Initializing Tari Pulse Service");
+        let shutdown_signal = context.get_shutdown_signal();
+        let (sender, receiver) = watch::channel(false);
+        context.register_handle(TariPulseHandle {
+            shutdown_signal: shutdown_signal.clone(),
+            failed_checkpoints_notifier: receiver,
+        });
+        let config = TariPulseConfig {
+            check_interval: self.interval.unwrap_or_default(),
```

Comment:
I suspect this will almost always be 0 and not the default of 120 you had it as.